test/e2e/agent/agent_test.go

```diff
@@ -63,4 +65,207 @@ var _ = Describe("VM Agent behavior", func() {
+			})
+
+			logrus.Infof("Waiting for the device to pick the config")
+			harness.WaitForDeviceContents(deviceId, "the device is upgrading to renderedVersion: 2",
+				func(device *v1alpha1.Device) bool {
+					for _, condition := range device.Status.Conditions {
+						if condition.Type == "Updating" && condition.Reason == "Updated" && condition.Status == "False" &&
+							condition.Message == UpdateRenderedVersionSuccess.String() {
+							return true
+						}
+					}
+					return false
+				}, TIMEOUT)
+
+			// The device should have the online config.
+			stdout, err := harness.VM.RunSSH([]string{"cat", "/etc/motd"}, nil)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(stdout.String()).To(ContainSubstring("This system is managed by flightctl."))
+
+			// The status should be "Online"
+			logrus.Infof("The device has the config %s", device.Spec.Config)
+			Eventually(harness.GetDeviceWithStatusSummary, TIMEOUT, POLLING).WithArguments(
+				deviceId).Should(Equal(v1alpha1.DeviceSummaryStatusType("Online")))
+		})
+
+		It("should report the correct device status when trying to upgrade to a not existing image", func() {
+			deviceId, _ := harness.EnrollAndWaitForOnlineStatus()
+
+			var newImageReference string
+			harness.UpdateDeviceWithRetries(deviceId, func(device *v1alpha1.Device) {
+				currentImage := device.Status.Os.Image
+				logrus.Infof("Current image for %s is %s", deviceId, currentImage)
+				repo, _ := parseImageReference(currentImage)
+				newImageReference = repo + ":not-existing"
+				device.Spec.Os = &v1alpha1.DeviceOSSpec{Image: newImageReference}
+				logrus.Infof("Updating %s to image %s", deviceId, device.Spec.Os.Image)
+			})
+
+			harness.WaitForDeviceContents(deviceId, "Failed to update to renderedVersion: 2. Retrying",
+				func(device *v1alpha1.Device) bool {
+					return conditionExists(device, "Updating", "True", "Retry")
+				}, "2m")
+
+			Eventually(harness.GetDeviceWithStatusSummary, TIMEOUT, POLLING).WithArguments(
+				deviceId).Should(Equal(v1alpha1.DeviceSummaryStatusType("Degraded")))
+		})
+
+		It(`should show an error when trying to update a device with
+			"a reference to a not existing git repo, and report 'Online' status`, Label("git"), func() {
+			deviceId, _ := harness.EnrollAndWaitForOnlineStatus()
+
+			harness.UpdateDeviceWithRetries(deviceId, func(device *v1alpha1.Device) {
+				logrus.Infof("Current device is %s", deviceId)
+
+				// Create ConfigProviderSpec.
+				var configProviderSpec v1alpha1.ConfigProviderSpec
+				err := configProviderSpec.FromGitConfigProviderSpec(gitConfigInvalidRepo)
+				Expect(err).ToNot(HaveOccurred())
+
+				device.Spec.Config = &[]v1alpha1.ConfigProviderSpec{configProviderSpec}
+				logrus.Infof("Updating %s with config %s", deviceId, device.Spec.Config)
+			})
+
+			// Check the http config error is detected.
+			harness.WaitForDeviceContents(deviceId, `1 invalid configuration: example-git-config-provider. Error: failed fetching
```

Comment:
```suggestion
			harness.WaitForDeviceContents(deviceId, `1 invalid configuration: example-git-config-provider. Error: failed fetching
```

That description should be shorter, it's going to make multi-line logs, it's something to help anybody reading the logs make sense of what's being waited for:

(( logrus.Infof("Waiting for condition: %q to be met", description) ))

Reply:
I updated the description, so the user message now would be: Waiting for condition: Error: failed fetching specified Repository definition to be met